### PR TITLE
Resolves ApcVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the ApcVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Power/APC/ApcVisualizerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namespace Content.Client.Power.APC;
+
+[RegisterComponent]
+[Access(typeof(ApcVisualizerSystem))]
+public sealed class ApcVisualizerComponent : Component
+{
+    public static readonly Color LackColor = Color.FromHex("#d1332e");
+    public static readonly Color ChargingColor = Color.FromHex("#2e8ad1");
+    public static readonly Color FullColor = Color.FromHex("#3db83b");
+    public static readonly Color EmagColor = Color.FromHex("#1f48d6");
```
Comment (AJCM-git) [inline Content.Client/Power/APC/ApcVisualizerComponent.cs:10]: i dont remember how the serializer handles static but try to make these datafields and assign the colors in the prototype instead (if nothing blows up test hot reloading) (also VV)

File: Content.Client/Power/APC/ApcVisualizerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Power.APC;
+
+public sealed class ApcVisualizerSystem : VisualizerSystem<ApcVisualizerComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ApcVisualizerComponent, ComponentInit>(OnInit);
```
Comment (AJCM-git) [inline Content.Client/Power/APC/ApcVisualizerSystem.cs:12]: oncomponentinit

File: Content.Client/Power/APC/ApcVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    ApcChargeState.Charging => ApcVisualizerComponent.ChargingColor,
+                    ApcChargeState.Full => ApcVisualizerComponent.FullColor,
+                    ApcChargeState.Emag => ApcVisualizerComponent.EmagColor,
+                    _ => ApcVisualizerComponent.LackColor
+                };
+            }
+        }
+        else
+        {
+            args.Sprite.LayerSetState(Layers.ChargeState, "apco3-0");
```
Comment (AJCM-git) [inline Content.Client/Power/APC/ApcVisualizerSystem.cs:87]: unhardocde

File: Content.Client/Power/APC/ApcVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+}
+
+enum Layers : byte
```
Comment (AJCM-git) [inline Content.Client/Power/APC/ApcVisualizerSystem.cs:92]: rename to ApcVisualLayers

File: Resources/Prototypes/Entities/Structures/Power/Generation/generators.yml [hunk 5]
```diff
@@ -108,8 +108,7 @@
     - state: on
       shader: unshaded
 #  - type: Appearance
-#    visuals:
-#    - type: ApcVisualizer
+#  - type: ApcVisualizer
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Structures/Power/Generation/generators.yml:111]: just remove this

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

**Comment by TemporalOroboros** [comment]
@DrSmugleaf Commit tested and pushed.

**Comment by TemporalOroboros** [comment]
Broken at the moment. Will fix it later.

**Comment by TemporalOroboros** [comment]
The APC screen works. The indicator lights should work, but the backend hasn't been implemented yet as it is out of scope for this PR. Should save some work for whoever implements them in the future.

File: Content.Client/Power/APC/ApcVisualizerComponent.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
+using Content.Shared.APC;
+
+namespace Content.Client.Power.APC;
+
+[RegisterComponent]
+[Access(typeof(ApcVisualizerSystem))]
+public sealed class ApcVisualsComponent : Component
```
Comment (metalgearsloth) [inline Content.Client/Power/APC/ApcVisualizerComponent.cs:7]: Shouldn't this use the component?

**Review by metalgearsloth — DISMISSED** [review]

File: Content.Client/Power/APC/ApcVisualizerComponent.cs (hunk 6 above)
Comment (AJCM-git) [reply Content.Client/Power/APC/ApcVisualizerComponent.cs:7]: The component is in the server and moving it to shared would require moving BaseNetConnectorComponent to shared as well as other power stuff which i think its out of scope for this pr

**Review by AJCM-git — CHANGES_REQUESTED** [review]
It does'nt work in-game

**Comment by TemporalOroboros** [comment]
> It does'nt work in-game

Can you elaborate on this? I have tested it and it appears to work as intended.

**Comment by AJCM-git** [comment]
> > It does'nt work in-game
> 
> Can you elaborate on this? I have tested it and it appears to work as intended.

Well when i tested the apc screen didnt seem to respond to changes in power, i deleted th MV cables going into it and the screen remained green, recharging also stayed green

**Comment by TemporalOroboros** [comment]
> > > It does'nt work in-game
> > 
> > 
> > Can you elaborate on this? I have tested it and it appears to work as intended.
> 
> Well when i tested the apc screen didnt seem to respond to changes in power, i deleted th MV cables going into it and the screen remained green, recharging also stayed green

The APC stays green if it's above 90% charge. It only goes green/blue if it's below that.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**